packages/cli/src/cmds/stats/stats.ts

```diff
@@ -1,335 +1,51 @@
 import yargs from 'yargs';
-import { chdir } from 'process';
-const fs = require('fs');
-import UI from '../userInteraction';
-import Telemetry from '../../telemetry';
-import { listAppMapFiles, verbose } from '../../utils';
-import { Event, buildAppMap } from '@appland/models';
-
-import { AppMapSize, AppMapSizeTable, SortedAppMapSize } from './types/appMapSize';
-import { FunctionExecutionTime, SlowestExecutionTime } from './types/functionExecutionTime';
-import chalk from 'chalk';
-import { relative } from 'path';
+import { statsForDirectory } from './directory/statsForDirectory';
+import { EventInfo, statsForMap } from './directory/statsForMap';
+import { SortedAppMapSize } from './types/appMapSize';
+import { SlowestExecutionTime } from './types/functionExecutionTime';
 
 export const command = 'stats [directory]';
-export const describe = 'Show some statistics about events in scenarios read from AppMap files';
+export const describe =
+  'Show statistics about events from an AppMap or from all AppMaps in a directory';
 const LIMIT_DEFAULT = 10;
-const MINIMUM_APPMAP_SIZE = (1024 * 1024) / 1;
 
 export const builder = (args: yargs.Argv) => {
   args.option('directory', {
-    describe: 'Working directory for the command.',
+    describe: 'Directory to analyze',
```

Comment:
Actually, the former description is correct. `--directory` changes the working directory of the command. This is used by many of our CLI commands. Please leave the comment as-is. 

If you want to specify a directory within `--directory`, add an `--appmap-dir` option, which by default will be ascertained from the _appmap.yml_. You'll see handling of these options in many other commands, which you can emulate.